### ISSUE TITLE
Windows Fatal Exception on Hyper Parameter Optimization

### ISSUE THREAD

**Issue by nielsuit227** [catboost/catboost#1729]
Problem:
During hyperparameter optimization, I got a Windows Fatal Exception with Optuna. I can reproduce the error rather easily without any other python module.  Interesting enough, it errors on the 9th iteration on a PC with 8 cpu cores. Reckon the other models are not properly cleaned or so. Am I doing something wrong or is this a bug?

catboost version: 0.25.1
Operating System: Windows 10 (OS Build 19042.985)
Python: 3.9.1
CPU: Intel i7 vPro 8th gen
# GPU: n/a

To reproduce:
```python
import copy
import catboost
import itertools
import numpy as np
from sklearn.datasets import make_classification


def test_cb():
    x, y = make_classification()
    model = catboost.CatBoostClassifier()
    for i, params in enumerate(get_params()):
        print(i, params)
        m = copy.deepcopy(model)
        m.set_params(**params)
        m.fit(x, y)


def get_params():
    params = {
        'verbose': [0],
        'loss_function': ['Logloss'],
        'allow_writing_files': [False],
        'early_stopping_rounds': [100],
        'learning_rate': np.logspace(-5, 0, 5).tolist(),
        'l2_leaf_reg': np.logspace(-5, 0, 5).tolist(),
        'depth': [3, 5, 7, 10],
        'min_data_in_leaf': [1, 50, 500, 1000],
        'grow_policy': ['SymmetricTree', 'Depthwise', 'Lossguide'],
    }
    k, v = zip(*params.items())
    return [dict(zip(k, v)) for v in itertools.product(*params.values())]


if __name__ == '__main__':
    test_cb()
    print('finished')
```

**Comment by nielsuit227** [comment]
So this was actually rather dumb on my side, but the error code was misleading and might be nice for future reference to be implemented. 
When the amount of samples is smaller than the `min_data_in_leaf`, and the `grow_policy` is `Depthwise`, which is for obvious reasons a bad idea, the CPU spikes and the process is killed.

**Comment by grag90** [comment]
Thank you for reporting the issue. Yes, the reason of this issue was value of `min_data_in_leaf`, which was higher than the size of dataset. This leads to segfault in cases when `grow_policy` is `Depthwise` or `Lossguide`, because catboost couldn't make any split of data. We will fix it.

**Comment by grag90** [comment]
Bug was fixed in 0.26.1.
